NFT/project/NFT-contract/src/interface.sw

```diff
@@ -2,247 +2,29 @@ library interface;
-    approved: Option<Identity>,
-    /// The user that has given or revoked approval to transfer his/her tokens.
-    owner: Identity,
-    /// The unique identifier of the token which the approved may transfer.
-    token_id: u64,
-}
-
-pub struct BurnEvent {
-    /// The user that has burned their token.
-    owner: Identity,
-    /// The unique identifier of the token which has been burned.
-    token_id: u64,
-}
-
-pub struct MintEvent {
-    /// The owner of the newly minted tokens.
-    owner: Identity,
-    /// The starting range of token ids that have been minted in this transaction.
-    token_id_start: u64,
-    /// The total number of tokens minted in this transaction.
-    total_tokens: u64,
-}
-
-abi NFT {
-    /// Returns the current admin for the contract.
-    ///
-    /// # Reverts
-    ///
-    /// * When the contract does not have an admin.
-    #[storage(read)]
-    fn admin() -> Identity;
-
-    /// Gives approval to the `approved` user to transfer a specific token on another user's behalf.
-    ///
-    /// To revoke approval the approved user should be `None`.
-    ///
-    /// # Arguments
-    ///
-    /// * `approved` - The user which will be allowed to transfer the token on the owner's behalf.
-    /// * `token_id` - The unique identifier of the token which the owner is giving approval for.
-    ///
-    /// # Reverts
-    ///
-    /// * When `token_id` does not map to an existing token.
-    /// * When the sender is not the token's owner.
-    #[storage(read, write)]
-    fn approve(approved: Identity, token_id: u64);
-
-    /// Returns the user which is approved to transfer the given token.
-    ///
-    /// If there is no approved user or the unique identifier does not map to an existing token,
-    /// the function will return `None`.
-    ///
-    /// # Arguments
-    ///
-    /// * `token_id` - The unique identifier of the token which the approved user should be returned.
-    ///
-    /// # Reverts
-    ///
-    /// * When there is no approved for the `token_id`.
-    #[storage(read)]
-    fn approved(token_id: u64) -> Identity;
-
-    /// Returns the balance of the `owner` user.
-    ///
-    /// # Arguments
-    ///
-    /// * `owner` - The user of which the balance should be returned.
-    #[storage(read)]
-    fn balance_of(owner: Identity) -> u64;
-
-    /// Burns the specified token.
-    ///
-    /// When burned, the metadata of the token is removed. After the token has been burned, no one
-    /// will be able to fetch any data about this token or have control over it.
-    ///
-    /// # Arguments
-    ///
-    /// * `token_id` - The unique identifier of the token which is to be burned.
-    ///
-    /// * Reverts
-    ///
-    /// * When `token_id` does not map to an existing token.
-    /// * When sender is not the owner of the token.
-    #[storage(read, write)]
-    fn burn(token_id: u64);
+use data_structures::TokenMetadata;
 
+abi Auxiliary {
```

Comment:
This isn't part of the core NFT ABI functionality and cannot be named NFT as it would conflict with the NFT ABI imported from the Sway-Libs repo. These are functions that are specific to this NFT and are not general functions. Not every NFT will have a `constructor()` nor will every NFT have the same struct returned from `token_metadata()`. 

I am terrible at naming so if there is something better suited here I'd love some input.